lib/puppet/provider/network_config/interfaces.rb

```diff
@@ -292,10 +289,24 @@ def self.format_file(filename, providers)
         stanza << "#{section} #{provider.send property}" if provider.send(property) and provider.send(property) != :absent
       end
 
+      if provider.mode == :vlan
+          if provider.options["vlan-raw-device"]
+            stanza << "vlan-raw-device #{provider.options["vlan-raw-device"]}"
+          else
+            vlan_range_regex = %r[\d{1,3}|40[0-9][0-5]]
```

Comment:
@vide can you explain this regex? If I'm reading it right it looks for three consecutive digits or four in a weird range between 4000-4095, excepting when the last digit is 6-9. Vlan 4006 should be valid, for instance. Then in your test you use vlan 4500 as an ID. What am I missing?